src/pages/instances/MigrateInstanceModal.tsx

```diff
@@ -1,197 +1,120 @@
-import { FC, KeyboardEvent, useState } from "react";
-import {
-  ActionButton,
-  Button,
-  MainTable,
-  Modal,
-} from "@canonical/react-components";
+import { FC, KeyboardEvent } from "react";
+import { Modal } from "@canonical/react-components";
 import { LxdInstance } from "types/instance";
-import { useQuery } from "@tanstack/react-query";
-import { queryKeys } from "util/queryKeys";
-import { fetchClusterMembers } from "api/cluster";
-import Loader from "components/Loader";
-import ScrollableTable from "components/ScrollableTable";
+import { useSettings } from "context/useSettings";
+import { isClusteredServer } from "util/settings";
+import FormLink from "components/FormLink";
+import ClusterMemberMigrationTable from "./ClusterMemberMigrationTable";
+import BackLink from "components/BackLink";
+import StoragePoolMigrationTable from "./StoragePoolMigrationTable";
+import { useInstanceMigration } from "util/instanceMigration";
 
 interface Props {
   close: () => void;
-  migrate: (target: string, instance: LxdInstance) => void;
   instance: LxdInstance;
 }
 
-const MigrateInstanceModal: FC<Props> = ({ close, migrate, instance }) => {
-  const { data: members = [], isLoading } = useQuery({
-    queryKey: [queryKeys.cluster, queryKeys.members],
-    queryFn: fetchClusterMembers,
-  });
-
-  const [selectedMember, setSelectedMember] = useState("");
+const MigrateInstanceModal: FC<Props> = ({ close, instance }) => {
+  const { data: settings } = useSettings();
+  const isClustered = isClusteredServer(settings);
+  const {
+    migrationType,
+    setMigrationType,
+    target,
+    setTarget,
+    pools,
+    members,
+    handleMigrate,
+  } = useInstanceMigration({ instance, close });
 
   const handleEscKey = (e: KeyboardEvent<HTMLElement>) => {
     if (e.key === "Escape") {
       close();
     }
   };
 
-  const handleMigrate = (instance: LxdInstance) => {
-    migrate(selectedMember, instance);
-    close();
-  };
+  const handleGoBack = () => {
+    // if lxd is not clustered, we close the modal
+    if (!isClustered) {
+      close();
+      return;
+    }
```

Comment:
```suggestion
      close();
```
I think this should move to the bottom of the function and we don't need the comment, because we can also cancel on the 1st step in the clustered case and want to close under that condition.

Reply:
`migrationType` would be set to "root storage pool" initially if lxd is not clustered. Moving `close` to the bottom would result in the modal not being able to close.

Reply:
Oh right, there is the special case for legacy. I don't really like that it increases complexity here, but don't have a better idea. Probably fine as is.

Do we need another `close()` at the end of the function for the first step in the clustered case?

Reply:
The first step for the clustered case does not have a cancel button. The modal at that point will be closed by `close()` always.